framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -518,24 +518,38 @@ protected void addInformationFromPreconditions(
                                 methodTree, method.getClassTree(), analysis.checker.getContext());
             }
 
+            TreePath localScope = analysis.atypeFactory.getPath(methodTree);
+
+            annotation = standardizeAnnotationFromContract(annotation, flowExprContext, localScope);
+
             try {
                 // TODO: currently, these expressions are parsed at the
                 // declaration (i.e. here) and for every use. this could
                 // be optimized to store the result the first time.
                 // (same for other annotations)
                 FlowExpressions.Receiver expr =
                         FlowExpressionParseUtil.parse(
-                                expression,
-                                flowExprContext,
-                                analysis.atypeFactory.getPath(methodTree),
-                                false);
+                                expression, flowExprContext, localScope, false);
                 info.insertValue(expr, annotation);
             } catch (FlowExpressionParseException e) {
                 // Errors are reported by BaseTypeVisitor.checkContractsAtMethodDeclaration()
             }
         }
     }
 
+    /** Standardize a type qualifier annotation obtained from a contract. */
+    private AnnotationMirror standardizeAnnotationFromContract(
+            AnnotationMirror annoFromContract,
+            FlowExpressionContext flowExprContext,
+            TreePath path) {
+        if (analysis.dependentTypesHelper != null) {
+            return analysis.dependentTypesHelper.standardizeAnnotation(
+                    flowExprContext, path, annoFromContract, false);
```

Comment:
Class `BaseTypeVisitor` also has a `standardizeAnnotationFromContract` method, which looks quite similar.
That other method also calls `dependentTypesHelper.checkAnnotation`.
Is such a call also necessary here? Would a comment help explain why or why not?
Cross-references between the two methods would make finding this similar logic easier.

It would be even nicer if that common logic were only implemented once.